src/toil/provisioners/abstractProvisioner.py

```diff
@@ -442,14 +443,16 @@ def __init__(self):
             # Holds strings like "ssh-rsa actualKeyData" for keys to authorize (independently of cloud provider's system)
             self.sshPublicKeys = []
 
-        def addFile(self, path: str, owner: str = 'root', permissions: str = '0755', content: str = ''):
+        def addFile(self, path: str, filesystem: str = 'root', mode: int = 755, contents: str = ''):
```

Comment:
`'0755'` is in octal right? It looks like Ignition expects decimal values (see `mode` in https://coreos.github.io/ignition/configuration-v3_2).

I'll suggest a change.

Reply:
Yep, I think it should be changed to a decimal value instead.